docs/proposals/2442-jax-runtime-trainer-v2/README.md

```diff
@@ -0,0 +1,176 @@
+| **Devices** | Compute units on a host (CPU cores, GPUs, TPUs). JAX detects devices automatically and runs parallel computations via `jax.pmap`, `jax.shard_map`, or `pjit`. Each JAX process accesses all devices on its host. **No need to spawn multiple processes per GPU** (unlike PyTorch). |
+| **Pod** | A Kubernetes Pod runs a single JAX process. Scheduled on a node and may use one or more GPUs depending on resource specifications. |
+| **Node** | A Kubernetes Node is a worker machine. In multi-node JAX jobs, each node typically runs one pod, mapping to one JAX host. |
+
+
+### JAX Training Workflow
+
+This section explains the architecture and flow of executing a distributed JAX training job using Kubeflow, as depicted in the diagram.
+
+![user-roles](./drawing.drawio.svg)
+
+| **Actor / Component**   | **Action**                                | **Details**                                                                                                                     |
+| ----------------------- | ----------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------- |
+| Platform Admin          | Prepares the Cluster Training Runtime | Defines container image, entrypoint, framework (e.g., JAX), and resource needs. Setup reusable for training jobs.               |
+| System                  | Retrieves the Training Runtime Spec   | Fetched automatically when a user requests a training job to determine execution details.                                       |
+| AI Practitioner         | Creates the Training Job              | Uses Kubeflow Python SDK or `kubectl`. Provides training function (e.g., `jax_train_mnist`), arguments, and node configuration. |
+| Runtime | Creates and Submits a JobSet          | Training job spec is translated into a JobSet (group of coordinated jobs).                                                      |
+| JobSet Controller | Launches Distributed Jobs             | JobSet spawns multiple Kubernetes Jobs, each pod runs a JAX training process instance.                                          |
+| Headless Service        | Connects Pods for Communication           | Enables direct pod-to-pod communication for gradient sharing and coordination.                                                  |
+| Cluster (Pods)    | Executes Distributed Training         | Each pod runs JAX+Python code, collaborating to complete training across available hardware.                                    |
+
+
+### Defining Distributed JAX with MLPolicy
+
+The number of **JAX hosts** is configured using the `numNodes` field in the **MLPolicy** section of the **ClusterTrainingRuntime**. Each host runs a single JAX process inside a Pod.
+
+#### JAXMLPolicySource
+
+`JAXMLPolicySource` configures JAX distributed initialization, backend, devices, and precision.
```

Comment:
Please adjust this text to say that `JAXMLPolicySource` API indicates that JAX plugin should be activated and extension framework will set the appropriate values for JAX distributed environment, etc.